homeassistant/components/rako/light.py

```diff
@@ -0,0 +1,212 @@
+        self._brightness = self._init_get_brightness_from_cache()
+        self._available = True
+
+    @property
+    def name(self) -> str:
+        """Return the display name of this light."""
+        raise NotImplementedError()
+
+    def _init_get_brightness_from_cache(self) -> int:
+        raise NotImplementedError()
+
+    async def async_added_to_hass(self) -> None:
+        """Run when entity about to be added to hass."""
+        await self.bridge.register_for_state_updates(self)
+
+    async def async_will_remove_from_hass(self) -> None:
+        """Run when entity about to be added to hass."""
+        await self.bridge.deregister_for_state_updates(self)
+
+    @property
+    def unique_id(self) -> str:
+        """Light's unique ID."""
+        return create_unique_id(
+            self.bridge.mac, self._light.room_id, self._light.channel_id
+        )
+
+    @property
+    def available(self) -> bool:
+        """Return True if entity is available."""
+        return self._available
+
+    @property
+    def brightness(self) -> int:
+        """Return the brightness of the light."""
+        return self._brightness
+
+    @brightness.setter
+    def brightness(self, value: int) -> None:
+        """Set the brightness. Used when state is updated outside Home Assistant."""
+        self._brightness = value
+        self.async_write_ha_state()
+
+    @property
+    def is_on(self) -> bool:
+        """Return true if light is on."""
+        return self.brightness > 0
+
+    @property
+    def should_poll(self) -> bool:
+        """Entity pushes its state to HA."""
+        return False
+
+    @property
+    def supported_features(self) -> int:
+        """Flag supported features."""
+        return SUPPORT_BRIGHTNESS
+
+    async def async_turn_off(self, **kwargs: Any) -> None:
+        """Turn off the light."""
+        await self.async_turn_on(brightness=0)
+
+    @property
+    def device_info(self) -> DeviceInfo:
+        """Return device information about this Rako Light."""
+        return {
+            "identifiers": {(DOMAIN, self.unique_id)},
+            "name": self.name,
+            "manufacturer": "Rako",
+            "suggested_area": self._light.room_title,
+            "via_device": (DOMAIN, self.bridge.mac),
+        }
+
+
+class RakoRoomLight(RakoLight):
+    """Representation of a Rako Room Light."""
+
+    def __init__(self, bridge: RakoBridge, light: python_rako.RoomLight) -> None:
+        """Initialize a RakoLight."""
+        super().__init__(bridge, light)
+        self._light: python_rako.RoomLight = light
+
+    def _init_get_brightness_from_cache(self) -> int:
+        scene_of_room = self.bridge.scene_cache.get(self._light.room_id, 0)
+        brightness: int = convert_to_brightness(scene_of_room)
+        return brightness
+
+    @property
+    def name(self) -> str:
+        """Return the display name of this light."""
+        room_title: str = self._light.room_title
+        return room_title
```

Comment:
It seems like the upstream library supports type hints, so this extra step of adding a local variable with an explicit type seems not necessary, so just inline. If the type can't be inferred then maybe improve ustream lib.

```suggestion
        return self._light.room_title
```

Reply:
(here and other places)